djangobulk/bulk.py

```diff
@@ -10,18 +10,41 @@
 from django.db import models, connections, transaction
 
 
+def _model_keys(model, field_names=None):
+    """Takes a model class and returns a list of fields that should be
+       used in a WHERE clause to an UPDATE.
+
+    :param model: A model class.
+    :param field_names: An iterable of field names that should be returned if
+        they match field in the model. If empty, no fields are returned.
+        If None the model's primary key is returned.
+    :returns: A list of fields that should be used in a WHERE
+    """
+    # If no fields are specified, use the primary key by default
+    if field_names is None:
+        return [model._meta.pk.name]
+    field_names = set(field_names)
+    fields = []
+    for f in model._meta.fields:
+        should_skip_field = field_names and f.name in field_names
```

Comment:
`field_names` contains the field names that should be returned. Here you are excluding them.
`should_keep_field = field_names and f.name in field_names`

Reply:
duh :)